allensdk/api/cloud_cache/cloud_cache.py

```diff
@@ -62,13 +191,41 @@ def latest_manifest_file(self) -> str:
         str
             the filename whose semver string is the latest one
         """
-        vstrs = [s.split(".json")[0].split("_v")[-1]
-                 for s in self.manifest_file_names]
-        versions = [semver.VersionInfo.parse(v) for v in vstrs]
-        imax = versions.index(max(versions))
-        return self.manifest_file_names[imax]
+        return self._find_latest_file(self.manifest_file_names)
+
+    @property
+    def latest_downloaded_manifest_file(self) -> str:
+        """parses downloaded available manifest files for semver string
+        and returns the latest one
+        self.manifest_file_names are assumed to be of the form
+        '<anything>_v<semver_str>.json'
+
+        Returns
+        -------
+        str
+            the filename whose semver string is the latest one
+        """
+        file_list = self.list_all_downloaded_manifests()
+        if len(file_list) == 0:
+            return ''
+        return self._find_latest_file(self.list_all_downloaded_manifests())
 
     def load_latest_manifest(self):
+        latest_downloaded = self.latest_downloaded_manifest_file
+        latest = self.latest_manifest_file
+        if latest != latest_downloaded:
+            if latest_downloaded != '':
+                msg = f'You are loading\n{self.latest_manifest_file}\n'
+                msg += 'which is newer than the most recent manifest '
+                msg += 'file you have previously been working with\n'
+                msg += f'{latest_downloaded}\n'
+                msg += 'It is possible that some data files have changed '
+                msg += 'between these two data releases, which will '
+                msg += 'force you to re-download those data files '
+                msg += '(currently downloaded files will not be overwritten).'
+                msg += f' To continue using {latest_downloaded}, run\n'
+                msg += f"self.load_manifest('{latest_downloaded}')"
```

Comment:
Same deal with `self` being potentially confusing.

Reply:
This has been done.

I've `self._user_interface_class` to `CloudCacheBase`, which defines the name of the class included in the warning messages. `VisualBehaviorOphysProjectCache` sets this variable when it instantiates its cloud-based `fetch_api`, which makes sure that the warning messages refer users to the proper aliases.